src/plugins/mer/merqmakebuildconfigurationfactory.cpp

```diff
@@ -0,0 +1,184 @@
+
+#include "merqmakebuildconfigurationfactory.h"
+#include "mersdkmanager.h"
+#include "merconstants.h"
+#include "mersdkkitinformation.h"
+
+#include <coreplugin/icore.h>
+#include <projectexplorer/target.h>
+#include <qtsupport/qtkitinformation.h>
+#include <qmakeprojectmanager/qmakeproject.h>
+#include <qmakeprojectmanager/qmakeprojectmanagerconstants.h>
+
+
+
+namespace {
+
+bool isFinished(const QProcess &p) {
+    return (p.exitStatus() == QProcess::NormalExit && p.exitCode() == 0);
+}
+
+QStringList readFrom(QProcess &p) {
+    return QString::fromUtf8(p.readAllStandardOutput()).split("\n\r", QString::SkipEmptyParts);
+}
+
+}
+
+
+
+using namespace ProjectExplorer;
+using namespace QmakeProjectManager;
+
+namespace Mer {
+namespace Internal {
+
+
+
+MerQmakeBuildConfigurationFactory::MerQmakeBuildConfigurationFactory():
+    QmakeBuildConfigurationFactory()
+{
+    registerBuildConfiguration<MerQmakeBuildConfiguration>(QmakeProjectManager::Constants::QMAKE_BC_ID);
+    setSupportedTargetDeviceTypes({Mer::Constants::MER_DEVICE_TYPE});
+    //setBasePriority(1);
+}
+
+
+
+int MerQmakeBuildConfigurationFactory::priority(const Kit *k, const QString &projectPath) const
+{
+    if (QmakeBuildConfigurationFactory::priority(k, projectPath) >= 0
+            && MerSdkManager::isMerKit(k))
+        return 1;
+    return -1;
+}
+
+
+
+int MerQmakeBuildConfigurationFactory::priority(const Target *parent) const
+{
+    const auto k = parent->kit();
+    if (k && QmakeBuildConfigurationFactory::priority(parent) >= 0
+            && MerSdkManager::isMerKit(k))
+        return 1;
+    return -1;
+}
+
+
+
+MerQmakeBuildConfiguration::MerQmakeBuildConfiguration(Target *target):
+    QmakeBuildConfiguration(target)
+{
+    const auto kit = target->kit();
+    const auto sdk = MerSdkKitInformation::sdk(kit);
+    const auto project = qobject_cast<QmakeProject *>(target->project());
+
+    Q_ASSERT(kit);
+    Q_ASSERT(sdk);
+    Q_ASSERT(project);
+
+
+    const QString mersshPath = Core::ICore::libexecPath() + QLatin1String("/merssh") + QStringLiteral(QTC_HOST_EXE_SUFFIX);
+
+    m_MerSsh.setProgram(QDir::toNativeSeparators(mersshPath));
+    m_MerSsh.setArguments({"qmake", "-env"});
+
+    //For the first time query build engine vars synchronously
```

Comment:
The build engine may not be running at this point or any later point where the query is run.